### ISSUE TITLE
Unable to use latest GitLab Helm chart with Kustomize and Argocd

### ISSUE THREAD

**Issue by MJDSys** [argoproj/argo-cd#13395]
Checklist:

* [x] I've searched in the docs and FAQ for my answer: https://bit.ly/argocd-faq.
* [x] I've included steps to reproduce the bug.
* [x] I've pasted the output of `argocd version`.

**Describe the bug**

I've created a  Kustomize file to deploy GitLab into my K8s cluster with Argocd.  With the latest GitLab Helm chart (6.11.0), Argocd is no longer able to produce the expected manifests.  I'm able to generate the template locally on my computer using the latest Kustomize release, this only occurs when run under the repo-server.  To eliminate any issues, I've tried a few different things:

1. I tried running Kustomize inside the Argocd docker container.  This was able to generate the manifests, so it's not just a version difference between my machine and Argocd
2. If I instead create a helm chart application in Argocd, then it is able to successfully create the templates.
3. Digging further, if I strace repo-server while it tries to create the template, I can see helm only accessing a single sub-chart's Chart.yaml file (gitlab-zoekt) with the broken GitLab Helm chart.  Doing the same thing with a working version of GitLab (ex. 6.10.0) shows Helm accessing all the Chart.yaml files.
4. The only significant difference I see with this sub-chart is that it includes a .git folder in the Chart file.  If I vendor a copy of the GitLab helm chart with this .git folder removed, then Argocd can generate the manifest.

I also tried this both on our deployed Argocd version (2.6.7) and the latest RC (2.7.0-rc2), with the same results.  This happens both in our production cluster (Kubernetes with CRIO running on FCOS) and locally using kind/docker.

**To Reproduce**

I've created a reproducer repository at: https://github.com/MJDSys/argocd-gitlab-reproducer .  This should be deployable directly into a new Argocd install.

**Expected behavior**

Argocd generates a template for Gitlab, and let's me sync that into the cluster.

**Screenshots**

![image](https://user-images.githubusercontent.com/2875429/235486056-2104d6c3-c6a8-4b5d-bb1d-a1016a9894c6.png)

**Version**

```shell
argocd: v2.4.8+844f79e
  BuildDate: 2022-07-29T17:23:00Z
  GitCommit: 844f79eb9d8f3ab96d4ce6f8df211c6093a660ba
  GitTreeState: clean
  GoVersion: go1.18.4
  Compiler: gc
  Platform: linux/amd64
argocd-server: v2.7.0-rc2+483d26b.dirty
```
Note: I created/deployed the application through the web gui.

**Logs**

Repo-server logs after a hard refresh is requested:
```
time="2023-05-01T16:18:45Z" level=info msg="git fetch origin --tags --force --prune" dir=/tmp/_argocd-repo/901f30cc-96e9-4f2d-9cde-2b8e156ab52e execID=1d0b5
time="2023-05-01T16:18:45Z" level=info msg=Trace args="[git fetch origin --tags --force --prune]" dir=/tmp/_argocd-repo/901f30cc-96e9-4f2d-9cde-2b8e156ab52e operation_name="exec git" time_ms=523.9340850000001
time="2023-05-01T16:18:45Z" level=info msg="git checkout --force 406e245979ba07f1f17f85e8ea97dc79a8331201" dir=/tmp/_argocd-repo/901f30cc-96e9-4f2d-9cde-2b8e156ab52e execID=14a92
time="2023-05-01T16:18:45Z" level=info msg=Trace args="[git checkout --force 406e245979ba07f1f17f85e8ea97dc79a8331201]" dir=/tmp/_argocd-repo/901f30cc-96e9-4f2d-9cde-2b8e156ab52e operation_name="exec git" time_ms=2.118128
time="2023-05-01T16:18:45Z" level=info msg="git clean -fdx" dir=/tmp/_argocd-repo/901f30cc-96e9-4f2d-9cde-2b8e156ab52e execID=fb58b
time="2023-05-01T16:18:45Z" level=info msg=Trace args="[git clean -fdx]" dir=/tmp/_argocd-repo/901f30cc-96e9-4f2d-9cde-2b8e156ab52e operation_name="exec git" time_ms=1.640385
time="2023-05-01T16:18:45Z" level=info msg="git rev-parse HEAD" dir=/tmp/_argocd-repo/901f30cc-96e9-4f2d-9cde-2b8e156ab52e execID=02a4a
time="2023-05-01T16:18:45Z" level=info msg="git rev-parse HEAD" dir=/tmp/_argocd-repo/901f30cc-96e9-4f2d-9cde-2b8e156ab52e execID=f44a2
time="2023-05-01T16:18:45Z" level=info msg=Trace args="[git rev-parse HEAD]" dir=/tmp/_argocd-repo/901f30cc-96e9-4f2d-9cde-2b8e156ab52e operation_name="exec git" time_ms=1.534152
time="2023-05-01T16:18:45Z" level=info msg=Trace args="[git rev-parse HEAD]" dir=/tmp/_argocd-repo/901f30cc-96e9-4f2d-9cde-2b8e156ab52e operation_name="exec git" time_ms=1.748301
time="2023-05-01T16:18:45Z" level=info msg="kustomize build /tmp/_argocd-repo/901f30cc-96e9-4f2d-9cde-2b8e156ab52e --enable-helm" dir= execID=f0ed6
time="2023-05-01T16:18:45Z" level=info msg="kustomize build /tmp/_argocd-repo/901f30cc-96e9-4f2d-9cde-2b8e156ab52e --enable-helm" dir= execID=a68b2
time="2023-05-01T16:18:45Z" level=error msg="`kustomize build /tmp/_argocd-repo/901f30cc-96e9-4f2d-9cde-2b8e156ab52e --enable-helm` failed exit status 1: Error: Error: Chart.yaml file is missing\n: unable to run: 'helm template gitlab /tmp/_argocd-repo/901f30cc-96e9-4f2d-9cde-2b8e156ab52e/charts/gitlab --namespace gitlab -f /tmp/kustomize-helm-2964441190/gitlab-kustomize-values.yaml' with env=[HELM_CONFIG_HOME=/tmp/kustomize-helm-2964441190/helm HELM_CACHE_HOME=/tmp/kustomize-helm-2964441190/helm/.cache HELM_DATA_HOME=/tmp/kustomize-helm-2964441190/helm/.data] (is 'helm' installed?)" execID=a68b2
time="2023-05-01T16:18:45Z" level=info msg=Trace args="[kustomize build /tmp/_argocd-repo/901f30cc-96e9-4f2d-9cde-2b8e156ab52e --enable-helm]" dir= operation_name="exec kustomize" time_ms=58.028498
time="2023-05-01T16:18:45Z" level=error msg="finished unary call with code Unknown" error="`kustomize build /tmp/_argocd-repo/901f30cc-96e9-4f2d-9cde-2b8e156ab52e --enable-helm` failed exit status 1: Error: Error: Chart.yaml file is missing\n: unable to run: 'helm template gitlab .charts/gitlab --namespace gitlab -f /tmp/kustomize-helm-2964441190/gitlab-kustomize-values.yaml' with env=[HELM_CONFIG_HOME=/tmp/kustomize-helm-2964441190/helm HELM_CACHE_HOME=/tmp/kustomize-helm-2964441190/helm/.cache HELM_DATA_HOME=/tmp/kustomize-helm-2964441190/helm/.data] (is 'helm' installed?)" grpc.code=Unknown grpc.method=GenerateManifest grpc.service=repository.RepoServerService grpc.start_time="2023-05-01T16:18:44Z" grpc.time_ms=982.861 span.kind=server system=grpc
time="2023-05-01T16:18:45Z" level=error msg="`kustomize build /tmp/_argocd-repo/901f30cc-96e9-4f2d-9cde-2b8e156ab52e --enable-helm` failed exit status 1: Error: Error: Chart.yaml file is missing\n: unable to run: 'helm template gitlab /tmp/_argocd-repo/901f30cc-96e9-4f2d-9cde-2b8e156ab52e/charts/gitlab --namespace gitlab -f /tmp/kustomize-helm-1115224636/gitlab-kustomize-values.yaml' with env=[HELM_CONFIG_HOME=/tmp/kustomize-helm-1115224636/helm HELM_CACHE_HOME=/tmp/kustomize-helm-1115224636/helm/.cache HELM_DATA_HOME=/tmp/kustomize-helm-1115224636/helm/.data] (is 'helm' installed?)" execID=f0ed6
time="2023-05-01T16:18:45Z" level=info msg=Trace args="[kustomize build /tmp/_argocd-repo/901f30cc-96e9-4f2d-9cde-2b8e156ab52e --enable-helm]" dir= operation_name="exec kustomize" time_ms=84.13553
time="2023-05-01T16:18:45Z" level=error msg="finished unary call with code Unknown" error="`kustomize build /tmp/_argocd-repo/901f30cc-96e9-4f2d-9cde-2b8e156ab52e --enable-helm` failed exit status 1: Error: Error: Chart.yaml file is missing\n: unable to run: 'helm template gitlab .charts/gitlab --namespace gitlab -f /tmp/kustomize-helm-1115224636/gitlab-kustomize-values.yaml' with env=[HELM_CONFIG_HOME=/tmp/kustomize-helm-1115224636/helm HELM_CACHE_HOME=/tmp/kustomize-helm-1115224636/helm/.cache HELM_DATA_HOME=/tmp/kustomize-helm-1115224636/helm/.data] (is 'helm' installed?)" grpc.code=Unknown grpc.method=GetAppDetails grpc.service=repository.RepoServerService grpc.start_time="2023-05-01T16:18:44Z" grpc.time_ms=1010.97 span.kind=server system=grpc
time="2023-05-01T16:18:46Z" level=info msg="git fetch origin --tags --force --prune" dir=/tmp/_argocd-repo/901f30cc-96e9-4f2d-9cde-2b8e156ab52e execID=07e3d
time="2023-05-01T16:18:46Z" level=info msg=Trace args="[git fetch origin --tags --force --prune]" dir=/tmp/_argocd-repo/901f30cc-96e9-4f2d-9cde-2b8e156ab52e operation_name="exec git" time_ms=501.69339099999996
time="2023-05-01T16:18:46Z" level=info msg="git checkout --force 406e245979ba07f1f17f85e8ea97dc79a8331201" dir=/tmp/_argocd-repo/901f30cc-96e9-4f2d-9cde-2b8e156ab52e execID=24479
time="2023-05-01T16:18:46Z" level=info msg=Trace args="[git checkout --force 406e245979ba07f1f17f85e8ea97dc79a8331201]" dir=/tmp/_argocd-repo/901f30cc-96e9-4f2d-9cde-2b8e156ab52e operation_name="exec git" time_ms=1.970636
time="2023-05-01T16:18:46Z" level=info msg="git clean -fdx" dir=/tmp/_argocd-repo/901f30cc-96e9-4f2d-9cde-2b8e156ab52e execID=8e849
time="2023-05-01T16:18:46Z" level=info msg=Trace args="[git clean -fdx]" dir=/tmp/_argocd-repo/901f30cc-96e9-4f2d-9cde-2b8e156ab52e operation_name="exec git" time_ms=1.815068
time="2023-05-01T16:18:46Z" level=info msg="git rev-parse HEAD" dir=/tmp/_argocd-repo/901f30cc-96e9-4f2d-9cde-2b8e156ab52e execID=a6094
time="2023-05-01T16:18:46Z" level=info msg=Trace args="[git rev-parse HEAD]" dir=/tmp/_argocd-repo/901f30cc-96e9-4f2d-9cde-2b8e156ab52e operation_name="exec git" time_ms=1.2718
time="2023-05-01T16:18:46Z" level=info msg="kustomize build /tmp/_argocd-repo/901f30cc-96e9-4f2d-9cde-2b8e156ab52e --enable-helm" dir= execID=2b0de
time="2023-05-01T16:18:46Z" level=error msg="`kustomize build /tmp/_argocd-repo/901f30cc-96e9-4f2d-9cde-2b8e156ab52e --enable-helm` failed exit status 1: Error: Error: Chart.yaml file is missing\n: unable to run: 'helm template gitlab /tmp/_argocd-repo/901f30cc-96e9-4f2d-9cde-2b8e156ab52e/charts/gitlab --namespace gitlab -f /tmp/kustomize-helm-2929164269/gitlab-kustomize-values.yaml' with env=[HELM_CONFIG_HOME=/tmp/kustomize-helm-2929164269/helm HELM_CACHE_HOME=/tmp/kustomize-helm-2929164269/helm/.cache HELM_DATA_HOME=/tmp/kustomize-helm-2929164269/helm/.data] (is 'helm' installed?)" execID=2b0de
time="2023-05-01T16:18:46Z" level=info msg=Trace args="[kustomize build /tmp/_argocd-repo/901f30cc-96e9-4f2d-9cde-2b8e156ab52e --enable-helm]" dir= operation_name="exec kustomize" time_ms=59.958827
time="2023-05-01T16:18:46Z" level=error msg="finished unary call with code Unknown" error="`kustomize build /tmp/_argocd-repo/901f30cc-96e9-4f2d-9cde-2b8e156ab52e --enable-helm` failed exit status 1: Error: Error: Chart.yaml file is missing\n: unable to run: 'helm template gitlab .charts/gitlab --namespace gitlab -f /tmp/kustomize-helm-2929164269/gitlab-kustomize-values.yaml' with env=[HELM_CONFIG_HOME=/tmp/kustomize-helm-2929164269/helm HELM_CACHE_HOME=/tmp/kustomize-helm-2929164269/helm/.cache HELM_DATA_HOME=/tmp/kustomize-helm-2929164269/helm/.data] (is 'helm' installed?)" grpc.code=Unknown grpc.method=GenerateManifest grpc.service=repository.RepoServerService grpc.start_time="2023-05-01T16:18:45Z" grpc.time_ms=948.55 span.kind=server system=grpc
```

**Comment by ldacey** [comment]
Just ran into this issue as well. I upgraded to ArgoCD 2.7.0 so I am not sure if that is what caused it or not though



rpc error: code = Unknown desc = Manifest generation error (cached): `kustomize build .argocd/apps/gitlab/overlays/production --enable-alpha-plugins --enable-helm` failed exit status 1: Error: Error: Chart.yaml file is missing : unable to run: 'helm template gitlab --namespace gitlab .argocd/apps/gitlab/overlays/production/charts/gitlab --values /tmp/kustomize-helm-2964921876/gitlab-kustomize-values.yaml --include-crds' with env=[HELM_CONFIG_HOME=/tmp/kustomize-helm-2964921876/helm HELM_CACHE_HOME=/tmp/kustomize-helm-2964921876/helm/.cache HELM_DATA_HOME=/tmp/kustomize-helm-2964921876/helm/.data] (is 'helm' installed?)

**Comment by MFietze** [comment]
It looks like the updated GitLab Helm chart version 6.11.1 fixed the issue, at least the application now syncs fine and has no errors afterwards.

**Comment by ldacey** [comment]
Hm - no luck for me. I had it working after I swapped to 6.10.5. I just upgraded to 6.11.1 again and face the same issue (ArgoCD is 2.7.1)

**Comment by MJDSys** [comment]
This appears to be fixed by PR #13001.  The issue was the gitlab-zoekt sub-chart was not being removed when the repo-server attempted to clean up the repository, causing helm to fail to pull the chart when next invoked by kustomize.  PR #13001 more forcefully cleans up the git directory, which ensures this folder is removed.

As this is related to the underlying solved issue #3683, I'm going to close this as complete.

**Comment by ldacey** [comment]
Appreciate the update!
